### ISSUE TITLE
Keras tensorflow : Get predictions and their associated ground truth labels afetr model.evaluate() or model.predict()

### ISSUE THREAD

**Issue by marineb30** [tensorflow/datasets#2500]
I am new to Tensorflow and Keras so the answer is perhaps  simple, but I have a batched and prefetched tensorflow dataset (of type `tf.data.TFRecordDataset`) which consists in images and their label (int type) , and I apply a classification model on it.

            `y_pred = model.predict(tf_test_dataset)`

I would like to compare classification results from ground truth for each image and put this in a dictionary. To do that I tried to create this dictionary with y_pred for each image, and its Ground truth label :

```
        iterator = iter(tf_test_dataset)
        result_dict = {}
        for image_number in range(len(y_pred)):
            result_dict[image_number]= \
                {
                "Score": y_pred[i],
                "GroundTruth": iterator.get_next()[1].numpy()
            }
```

But I am not sure if it is the right way to get predictions and their associated ground truth label. (Is it easier using model.evaluate ?). Moreover when I do 

```
    iterator = iter(tf_test_dataset)
    print(iterator.get_next()[1].numpy())
```

and 

```
    for i in tf_test_dataset:
       print(i[1].numpy())
```

the results (the lists with the labels) are different... 

Thanks for your help

**Comment by Conchylicultor** [comment]
Are you using TFDS (`import tensorflow_datasets`) or `tf.data.Dataset` ?

If the labels are different, it might means your pipeline is non-deterministic (e.g. you're using `ds = ds.shuffle()`).

To get the prediction, you could use something like:

```py
for image, target in ds:
  y = model(image, training=False)  # Or y = model.predict(image)
  # then compare y with target
```
